llvm/lib/Target/AMDGPU/VOP1Instructions.td

```diff
@@ -729,7 +729,20 @@ def V_ACCVGPR_MOV_B32 : VOP1_Pseudo<"v_accvgpr_mov_b32", VOPProfileAccMov, [], 1
   let isAsCheapAsAMove = 1;
 }
 
+def VOP_SWAP_I16 : VOPProfile_True16<VOP_I16_I16> {
+  let Outs32 = (outs VOPDstOperand_t16Lo128:$vdst,
+                     VOPSrcEncodedDstOperand_t16Lo128:$vdst1);
+  let Ins32 = (ins VOPSrcEncodedDstOperand_t16Lo128:$src0,
+                   VOPDstOperand_t16Lo128:$src1);
+  let Asm32 = " $vdst, $src0";
+}
+
 let SubtargetPredicate = isGFX11Plus in {
+  def V_SWAP_B16 : VOP1_Pseudo<"v_swap_b16", VOP_SWAP_I16, [], /* VOP1Only= */ 1> {
```

Comment:
Are you sure this opcode is VOP1 only? When I implemented support for `v_swap_b16` in our compiler, I tested it with VOP3 encoding and it worked fine. VOP3 v_swap_b16 is nessecary to address v128-v255.

Reply:
According to the docs this should be VOP1 only for GFX11/12. I am not sure if such usage is supported when VOP3 is not listed in supported format.  Hi @Sisyph @kosarev Do you know anything about this?

Reply:
It's VOP1 only in GFX11 and GFX12, and it does support v0-v255 for both operands.

Reply:
That can't be true, VOP1 v_swap_b16  definitely operates on v0-v127.lo/.hi. There is no way to encode v128-v255 without VOP3, just like for other true16 instructions.

Reply:
Right, the lo/hi selector goes to the MSB of the operand fields, so can't encode registers beyond v127.

Reply:
This is what I tested for what it's worth, and that instruction swaps v138.h and v139.h like expected:

https://gitlab.freedesktop.org/DadSchoorse/bvhre/-/blob/d1a4108b1547577860261d39a24ca4ea616c7664/ptest.cpp#L265

Reply:
From what I see, officially there's no VOP3 form, but I could ask ISA people to be sure. Which targets did you try this on, exactly?

Reply:
I did all my gfx11 testing on navi31.

Reply:
OK, I did inquire on this and will update this thread as soon as I have a response. Thanks for reporting.

Reply:
The response from our HW guys reads that the VOP3 codes are not guaranteed to work correctly under all possible circumstances. Meaning, they should not be considered supported regardless of observed side effects.

Reply:
Thanks for investigating, even if the answer is a bit disappointing.